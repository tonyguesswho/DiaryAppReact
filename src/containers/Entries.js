import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import moment from 'moment';
import { getEntries } from '../actions/entriesActions';
import { signout } from '../actions/authAction';

class Entries extends Component {
  constructor() {
    super();
    this.state = {
      entries: [],
      loading: true
    };
  }
  logout = () => {
    signout();
  };
  componentDidMount = () => {
    const { getEntries } = this.props;
    getEntries();
  };
  componentWillReceiveProps(nextProps) {
    if (nextProps) {
      this.setState({
        entries: nextProps.entries,
        loading: false
      });
    }
  }
  render() {
    const { entries, loading } = this.state;
    return (
      <div className="bg body">
        <header className="header2">
          <nav>
            <ul className="nav nav2">
              <li>
                <Link to="/">Home</Link>
              </li>
              <li>
                <Link to="/" id="logout" onClick={this.logout}>
                  Sign Out
                </Link>
              </li>
            </ul>
          </nav>
        </header>
        {loading ? (
          <div>loading ...</div>
        ) : (
          <section className="entries">
            {Object.values(entries).length === 0 ? (
              <h2 id="ent"><p>No diary entry : <Link to="/create">Write</Link></p></h2>
            ) : (
              <div>
                <div className="bigbox" id="allEntries">
                  {Object.values(entries).map(entry => (
                    <div className="box card span41" id="anEntry" key={entry.id}>
                      <h4>{entry.title}</h4>
                      <Link to={`/entry/${entry.id}`} className="ebtn ebtn-filled">
                        View
                      </Link>
                      <Link to={`/editentry/${entry.id}`} className="ebtn ebtn-transparent">
                        Edit
                      </Link>
                      <div className="entry-date">
                        <p>{moment(`${entry.created_at}`).format('MMMM Do YYYY')}</p>
                      </div>
                    </div>
                  ))}
                </div>
              </div>
            )}
          </section>
        )}
      </div>
    );
  }
}
const mapStateToProps = state => ({ entries: state.entries.entries });
export default connect(
  mapStateToProps,
  { getEntries, signout }
)(Entries);
