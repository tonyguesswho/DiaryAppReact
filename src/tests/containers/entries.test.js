import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import Entries from '../../containers/Entries';
import Root from '../../root';

let wrapped;

beforeEach(() => {
  wrapped = mount(
    <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
      <Root>
        <Entries />
      </Root>
    </MemoryRouter>
  );
});

afterEach(() => wrapped.unmount());

describe('entry UI', () => {
  describe('render features', () => {
    test('container should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});
