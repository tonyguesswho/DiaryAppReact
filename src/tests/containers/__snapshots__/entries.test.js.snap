// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entry UI render features container should render as expected 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <_default>
      <Provider
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <Connect(Entries)>
          <Entries
            entries={Array []}
            getEntries={[Function]}
            signout={[Function]}
          >
            <div
              className="bg body"
            >
              <header
                className="header2"
              >
                <nav>
                  <ul
                    className="nav nav2"
                  >
                    <li>
                      <Link
                        replace={false}
                        to="/"
                      >
                        <a
                          href="/"
                          onClick={[Function]}
                        >
                          Home
                        </a>
                      </Link>
                    </li>
                    <li>
                      <Link
                        id="logout"
                        onClick={[Function]}
                        replace={false}
                        to="/"
                      >
                        <a
                          href="/"
                          id="logout"
                          onClick={[Function]}
                        >
                          Sign Out
                        </a>
                      </Link>
                    </li>
                  </ul>
                </nav>
              </header>
              <div>
                loading ...
              </div>
            </div>
          </Entries>
        </Connect(Entries)>
      </Provider>
    </_default>
  </Router>
</MemoryRouter>
`;
